third_party/closure-compiler/runner/src/AmpCommandLineRunner.java

```diff
@@ -0,0 +1,34 @@
+import com.google.common.collect.ImmutableSet;
+import com.google.javascript.jscomp.CommandLineRunner;
+import com.google.javascript.jscomp.CompilerOptions;
+import com.google.javascript.jscomp.CustomPassExecutionTime;
+
+public class AmpCommandLineRunner extends CommandLineRunner {
+  
+  ImmutableSet<String> suffixTypes = ImmutableSet.of(
+      "log.dev.fine",
+      "log.dev.assert"
+      );
+
+  protected AmpCommandLineRunner(String[] args) {
+    super(args);
+  }
+
+  @Override
+  protected CompilerOptions createOptions() {
+    CompilerOptions options = super.createOptions();
+    AmpPass ampPass = new AmpPass(getCompiler(), suffixTypes);
+    options.addCustomPass(CustomPassExecutionTime.BEFORE_OPTIMIZATIONS, ampPass);
```

Comment:
@cramforce see my commend here https://github.com/ampproject/amphtml/pull/2593#discussion_r56468684

unfortunately the `stripPrefix` (whether by `type` or `name`) operate on the module prefixed variable (guessing the es6 to es5 transpilation) so for it to strip the prefix type "dev.log" for example you would have to need to know it is actually named "module$__$__$__$extensions$amp_test$0_1$log.dev" during that phase (possible?). `name` would work but it doesn't allow for "GETPROP" (dot operator) so you can only give it "dev" or "log" for example which feels very unsafe.